amqp/src/main/java/io/github/tcdl/msb/adapters/amqp/AmqpAcknowledgementHandler.java

```diff
@@ -0,0 +1,77 @@
+package io.github.tcdl.msb.adapters.amqp;
+
+import io.github.tcdl.msb.api.AcknowledgementHandler;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rabbitmq.client.Channel;
+
+/**
+ * {@link AmqpAcknowledgementHandler} provides acknowledgement for AMQP broker. 
+ * Used from {@link AmqpMessageConsumer}.
+ */
+
+public class AmqpAcknowledgementHandler implements AcknowledgementHandler {
+
+    private static final String ACK_WAS_ALREADY_SENT = "Acknowledgement was already sent during message processing.";
+
+    private static final Logger LOG = LoggerFactory.getLogger(AmqpAcknowledgementHandler.class);
+
+    final Channel channel;
+    final String consumerTag;
+    final long deliveryTag;
+    final boolean isRequeueRejectedMessages;
+
+    boolean isAcknowledgementSent = false;
+
+    public AmqpAcknowledgementHandler(Channel channel, String consumerTag, long deliveryTag,
+            boolean isRequeueRejectedMessages) {
+        super();
+        this.channel = channel;
+        this.consumerTag = consumerTag;
+        this.deliveryTag = deliveryTag;
+        this.isRequeueRejectedMessages = isRequeueRejectedMessages;
+    }
+
+    @Override
+    public void confirmMessage() {
+        if (!isAcknowledgementSent) {
+            try {
+                channel.basicAck(deliveryTag, false);
+                isAcknowledgementSent = true;
+            } catch (Exception e) {
+                LOG.error(String.format("[consumer tag: %s] Got exception when trying to confirm a message:", consumerTag), e);
+            }
+        } else {
+            LOG.warn(ACK_WAS_ALREADY_SENT);
```

Comment:
I agree that just logging warning for the duplicate confirm is ok. But what if developer invoked `confirm` and then `reject`. I think it clearly indicates some problem in microservice logic and we should have some more evidence in logs then just a WARN message